### PR TITLE
ft(edit request): update request

### PR DESCRIPTION
**Pull request by Alexander96779**
#### What does this PR do?
This PR allows users to edit trip requests that are still open.
#### Description of Task to be completed?
- A user creates a trip request.
- And decides to change some information.
- He searches for that trip request record he/she wants to edit or modify.
- And then he can update or modify that trip request trip.
- Added tests
#### How should this be manually tested?
- You have to clone the repo `git clone https://github.com/Stackup-Rwanda/fan7-bn-backend.git`
- Then After the repo cloned `cd project-folder` and run `npm i` to install all dependencies
- Go to my branch `git checkout ft-edit-request-170764600`
- Run app by `npm run dev`
- Use Postman to test this endpoint, select `PATCH` API method and then use this URL `http://localhost:5000/api/requests/1/edit`, then choose the data you want to change and send the request.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#170764600](https://www.pivotaltracker.com/story/show/170764600)
#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by niyodusengaclement — APPROVED** [review]
ready to go

File: src/controllers/requestController.js [hunk 1]
```diff
@@ -292,6 +292,51 @@ class RequestController {
...
+    });
+    if (!request) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Request not found',
+      });
+    }
+    if (request.user_id === user.id) {
+      if (request.status === 'Pending') {
+        await RequestRepository.update({ id: requestId }, requestData);
```
Comment (istafrica-dennw) [inline src/controllers/requestController.js]: @Alexander96779  review this line 324 to 329 to catch errors

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

**Review by ajimae — APPROVED** [review]

**Review by ajimae — COMMENTED** [review]
Please implement your feedback

File: src/controllers/requestController.js (hunk 1 above)
Comment (Alexander96779) [reply src/controllers/requestController.js]: I am done

**Review by Skemc — APPROVED** [review]
Check your test

**Comment by Skemc** [comment]
Check your test

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**